### ISSUE TITLE
Returned result.files.single.path is NULL

### ISSUE THREAD

**Issue by pulstar** [miguelpruivo/flutter_file_picker#471]
**Describe the bug**

Running the following code in Flutter web:

```
    FilePickerResult result = await FilePicker.platform.pickFiles(
      type: fileType, allowedExtensions: extensions,
    );
    print ('result.files.single: ${result.files.single}');
    print ('result.files.single.name: ${result.files.single.name}');
    print ('result.files.single.path: ${result.files.single.path}');
    return (result != null) ? File(result.files.single.path) : null;
```

I got:

```
result.files.single: Instance of 'PlatformFile'
js_primitives.dart:30 result.files.single.name: EiHs8QXUcAAFQr8.jpg
js_primitives.dart:30 result.files.single.path: null
```

**Issue details**
1. Platform: Web
2. Platform OS version: Chrome under Windows 10
3. File type (filter): images

**Error Log**

`Invalid argument(s) (path): Must not be null`

**Flutter Version details**
```
[√] Flutter (Channel dev, 1.24.0-3.0.pre, on Microsoft Windows [version 10.0.18363.1139], locale pt-BR)
    • Flutter version 1.24.0-3.0.pre at C:\src\flutter
    • Framework revision 2783f8e2e1 (7 days ago), 2020-10-22 09:36:06 -0700
    • Engine revision defa8be2b1
    • Dart version 2.11.0 (build 2.11.0-242.0.dev)

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\pulst\AppData\Local\Android\Sdk
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = C:\Users\pulst\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[X] Visual Studio - develop for Windows
    X Visual Studio not installed; this is necessary for Windows development.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] Connected device (5 available)
    • Vivo XI (mobile)  • PRUCHYWGG6HYSK4H • android-arm64  • Android 9 (API 28)
    • Windows (desktop) • windows          • windows-x64    • Microsoft Windows [version 10.0.18363.1139]
    • Web Server (web)  • web-server       • web-javascript • Flutter Tools
    • Chrome (web)      • chrome           • web-javascript • Google Chrome 86.0.4240.111
    • Edge (web)        • edge             • web-javascript • Microsoft Edge 86.0.622.48

```

**Additional context**

May this problem have anything to do with **dart:io** ?

**Comment by miguelpruivo** [comment]
Hi @pulstar, like [mentioned on the note 2 at the wiki](https://github.com/miguelpruivo/flutter_file_picker/wiki/API#methods), paths aren't supported on web since browsers always use fake paths. You should use `bytes` property instead for retrieving the picked file.

Thank you!

**Comment by pulstar** [comment]
Thank you for the fast reply! 

Since I need a file to upload it to my server, I used the [File](https://pub.dev/packages/file) package and the code below to create it in memory:

```
File file = MemoryFileSystem().file(result.files.single.name)
        ..writeAsBytesSync(result.files.single.bytes);
```

But this approach didn't work fine with Firebase Storage, so I am looking forward for a better solution.

**Comment by miguelpruivo** [comment]
@pulstar unfortunately Firebase Storage doesn't support Flutter Web yet. But there are some people that actually came up with a solution, check [here](https://stackoverflow.com/questions/59716944/flutter-web-upload-image-file-to-firebase-storage).

**Comment by pulstar** [comment]
> @pulstar unfortunately Firebase Storage doesn't support Flutter Web yet. But there are some people that actually came up with a solution, check [here](https://stackoverflow.com/questions/59716944/flutter-web-upload-image-file-to-firebase-storage).

Amazing! Miguel, you rock! Thank you!

**Comment by pulstar** [comment]
I found this interesting article about this:

https://medium.com/digital-tech-articles/firebase-storage-in-flutter-web-d0d223d2b51a

**Comment by miguelpruivo** [comment]
@pulstar looks that it may contain relevant info. I might take a look. Thank you for the input.

**Comment by pulstar** [comment]
@miguelpruivo I implemented the above solution in my project, but I got the following error: 

`MissingPluginException(No implementation found for method getApplicationDocumentsDirectory on channel plugins.flutter.io/path_provider). `

I have no idea how to solve this... Instead of image_picker_web, I am using flutter_file_picker to get the file, because I want to upload any kind of files to my Firebase Storage. I did try the solution you gave above, but the IDE complained that the universal_html.File is not of the same type of dart:io.File... I am not sure what is wrong.

By the way, I am refactoring a lot of code to make my app fully web and mobile compatible. When I solve this, I will also try to replace my old image_crop plugin to this one: https://pub.dev/packages/crop which is web compatible.

**Comment by pulstar** [comment]
After a lot of try and error, I gave up the solutions we posted here. None works as expected or are fully compatible with the official FlutterFire packages, giving a lot of conflicts and errors. There is no official storage web package for Flutter, despite the fact that we already have the client library (https://www.gstatic.com/firebasejs/7.22.1/firebase-storage.js) and it runs in another Javascript applications (https://firebase.google.com/docs/storage/web/upload-files). This should work already, but it doesn't. So I will try to create a Firebase function to get the file and upload it to a storage bucket. I will need to deal with CORS and things like that, but it may work.
